src/components/landing/pages/register/register.tsx

```diff
@@ -0,0 +1,156 @@
+import React, { FormEvent, useEffect, useState } from 'react'
+import { useSelector } from 'react-redux'
+import { Redirect } from 'react-router'
+import { doInternalRegister } from '../../../../api/auth'
+import { ApplicationState } from '../../../../redux'
+import { getAndSetUser } from '../../../../utils/apiUtils'
+import { Row, Col, Card, Form, Button, Alert } from 'react-bootstrap'
+import { Trans, useTranslation } from 'react-i18next'
+import { TranslatedInternalLink } from '../../../common/links/translated-internal-link'
+import { ShowIf } from '../../../common/show-if/show-if'
+
+enum RegisterError {
+  NONE,
+  USERNAME_EXISTING,
+  OTHER
+}
+
+export const Register: React.FC = () => {
+  const { t } = useTranslation()
+  const config = useSelector((state: ApplicationState) => state.backendConfig)
+  const user = useSelector((state: ApplicationState) => state.user)
+
+  const [username, setUsername] = useState('')
+  const [password, setPassword] = useState('')
+  const [passwordAgain, setPasswordAgain] = useState('')
+  const [error, setError] = useState(RegisterError.NONE)
+  const [ready, setReady] = useState(false)
+
+  const doAsyncRegister = async (): Promise<void> => {
+    await doInternalRegister(username, password)
+    await getAndSetUser()
+  }
+
+  const doRegisterSubmit = (event: FormEvent): void => {
+    doAsyncRegister().catch((err: Error) => {
+      if (err.message === 'Username is already existing') {
```

Comment:
maybe we should avoid such magic strings and use the RegisterError  enum directly?

Reply:
Done in 8a38f328